src/Stratis.Bitcoin.Tests/P2P/PeerAddressManagerTests.cs

```diff
@@ -167,5 +189,104 @@ public void PeerAddressManager_AttemptThresholdTimeNotReached_DoNotReset()
+
+            // Act
+            addressManager.AddPeer(endpoint, IPAddress.Loopback);
+
+            // Assert - Peer does not exist in the store as banned.
+            PeerAddress savedPeer = addressManager.FindPeer(endpoint);
+            Assert.Null(savedPeer);
+        }
+
+        [Fact]
+        public void PeerAddressMananger_BannedPeers_ShouldNotAppearInStore_AddPeer()
+        {
+            var goodIPAddress = @"::ffff:192.170.0.1";
+
+            // Arrange - Setup Peer and ban it using the PeerAddressManager
+            IPAddress ipAddress = IPAddress.Parse("::ffff:192.168.0.1");
+            var bannedEndPoint = new IPEndPoint(ipAddress, 80);
+            PeerAddressManager addressManager = this.PeerAddressManager(ipAddress, bannedEndPoint);
+
+            // Add good Peer
+            ipAddress = IPAddress.Parse(goodIPAddress);
+            var goodEndpoint = new IPEndPoint(ipAddress, 80);
+            addressManager.AddPeer(goodEndpoint, ipAddress);
+
+            // Act
+            addressManager.SavePeers();
+            addressManager.LoadPeers();
+
+            // Assert - Peer does not exist in the store as banned.
+            PeerAddress bannedPeer = addressManager.FindPeer(bannedEndPoint);
+            Assert.Null(bannedPeer);
+
+            // Assert - Good Peer does exist in the store
+            PeerAddress goodPeer = addressManager.FindPeer(goodEndpoint);
+            Assert.Equal(goodIPAddress, goodPeer.Endpoint.Address.ToString());
+        }
+
+        [Fact]
+        public void PeerAddressMananger_BannedPeers_ShouldNotAppearInStore_AddPeers()
+        {
+            // Arrange - Setup Peer and ban it using the PeerAddressManager
+            IPAddress ipAddress = IPAddress.Parse("::ffff:192.168.0.1");
+            var bannedEndPoint = new IPEndPoint(ipAddress, 80);
+            PeerAddressManager addressManager = this.PeerAddressManager(ipAddress, bannedEndPoint);
+
+            // Add banned peer and two good peers.
+            var peersToAdd = new List<IPEndPoint>()
+            {
+                bannedEndPoint,
+                new IPEndPoint(IPAddress.Parse("::ffff:192.168.0.3"), 80),
+                new IPEndPoint(IPAddress.Parse("::ffff:192.168.0.4"), 80),
+            };
+
+            addressManager.AddPeers(peersToAdd.ToArray(), IPAddress.Loopback);
+
+            // Act
+            addressManager.SavePeers();
+            addressManager.LoadPeers();
+
+            // Assert - Peer does not exist in the store as banned.
+            PeerAddress bannedPeer = addressManager.FindPeer(bannedEndPoint);
+            Assert.Null(bannedPeer);
+
+            // Assert - Good Peers exist in the store
+            var peers = addressManager.PeerSelector.SelectPeersForGetAddrPayload(3);
+            Assert.Equal(2, peers.Count());
+        }
+
+        private PeerAddressManager PeerAddressManager(IPAddress ipAddress, IPEndPoint endpoint)
```

Comment:
Maybe having a more representative method name would be better. Smth like `CreateMockedPeerAddressManager()`

Reply:
When I redo the implementation around this and if I still need this method I'll rename.  Cheers.